### ISSUE TITLE
no-floating-promises false positive (TSLint 5.20.0)

### ISSUE THREAD

**Issue by vladeck** [palantir/tslint#4867]
### Bug Report

- __TSLint version__: 5.20.0
- __TypeScript version__: 3.6.3
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
const func = async (): Promise<number> => new Promise(resolve => resolve(5));

const wrapper: {isBusy: boolean; response?: number; error?: unknown} = {
    isBusy: false
};

// Flagged as 'no-floating-promises'
func()
    .then(response => wrapper.response = response)
    .catch(e => wrapper.error = e)
    .finally(() => wrapper.isBusy = false);

// Not flagged
func()
    .finally(() => wrapper.isBusy = false)
    .then(
        response => wrapper.response = response,
        e => wrapper.error = e
    );
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:all",
    "rules": {}
}
```

**NOTE:** Adding `.catch` flags the code with `no-floating-promises` no matter the order.

#### Actual behavior
`no-floating-promises` error.

#### Expected behavior
No error.

**Comment by JoshuaKGoldberg** [comment]
@vladeck that's intentional, yes. The `.finally` code might itself throw an error.

```ts
func()
    .then(() => { /* ... */ })
    .catch(() => { /* ... */ })
    .finally(() => { throw new Error("aha!"); });
```

**Comment by vladeck** [comment]
Is this behavior intentional as well? If so, why?:

```ts
// Not flagged
func()
    .finally(() => { throw Error("?"); })
    .then(
        response => wrapper.response = response,
        e => wrapper.error = e
    );

// Not flagged
func()
    .then(
        response => wrapper.response = response,
        e => wrapper.error = e
    )
    .finally(() => { throw Error("?"); })
```

**Comment by JoshuaKGoldberg** [comment]
Aha, my apologies @vladeck, you're totally right - this looks like a bug.

Keeping #4534 in mind, this might be better served as a link to [typescript-eslint](https://typescript-eslint.io).

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
